code-challenges.test.js

```diff
@@ -13,47 +13,147 @@
+//create a function named 'divisibleBy3' that takes an object as an argument.
+const divisibleBy3 = (object) => {
+  //If the number found within the object is divisible by 3 return, use string
+  //interpolation to state that it is divisible by 3.
+  if (object.number % 3 === 0){
+    return `${object.number} is divisible by three`
+
+    //If the number found within the object is not divisible by 3 return, use string
+    //interpolation to state that it is not divisible by 3.
+  } else if (object.number % 3 !== 0){
+    return `${object.number} is not divisible by three`
+
+    //If all ifs fails, something went wrong
+  } else {
+    return 'Something wwent wrong'
+  }
+}
 
 // --------------------2) Create a function that takes in an array of words and returns an array with all the words capitalized.
 
 // a) Create a test with expect statements for each of the variables provided.
-
-const randomNouns1 = ["streetlamp", "potato", "teeth", "conclusion", "nephew"]
-// Expected output: ["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew"]
-const randomNouns2 = ["temperature", "database", "chopsticks", "mango"]
-// Expected output: ["Temperature", "Database", "Chopsticks", "Mango"]
+//Create a test for a function named variable.
+describe('capitalized', () => {
+
+  //Explain how the function is supposed to perform.
+  it("takes in an array of words and returns an array with all the words capitalized", () => {
+
+    //Variables that are used for the expect.
+    const randomNouns1 = ["streetlamp","potato","teeth","conclusion","nephew",];
+    // Expected output: ["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew"]
+    const randomNouns2 = ["temperature", "database", "chopsticks", "mango"];
+    // Expected output: ["Temperature", "Database", "Chopsticks", "Mango"]
+    
+    //hardcode what you would pass as an argument and what the 
+    //expected outcomes will be.
+    expect(capitalized(randomNouns1)).toEqual(["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew"])
+    expect(capitalized(randomNouns2)).toEqual(["Temperature", "Database", "Chopsticks", "Mango"])
+  })
+})
 
 // b) Create the function that makes the test pass.
+//create a function named 'capitalized' that takes an array as an argument. Use the .map() method to iterate through
+//the array. Use the .charAt() method to take the first character of each value which in this case would be the first
+//letter of each word - set that character to uppercase using .toUpperCase() method. add the additional characters of
+//each value using the .slice() method. We use .sclice(1) to take the remaining characters of each word and return a new
+//array with all the words capitalized.
+const capitalized = (array) => {return array.map((value) => value.charAt(0).toUpperCase()+value.slice(1))}
+//Note: this one was especially difficult to figure out. If there is an easier to understand method please
+//let me know. looking at it makes my eyes hurt.
```

Comment:
I think this is a great approach and it makes that there is a few steps to it because we can't just throw toUpperCase on the array and call it good. One thing I would consider is dropping it into a few more lines so it's easier to read but the logic and pseudocode is spot on!